File: test/zarr_probe/check.py
#!/usr/bin/env python3
"""Ordering check via xarray/zarr (spec §4.2) — authoritative for numpy consumers."""
import sys

import numpy as np
import xarray as xr

path = sys.argv[1] if len(sys.argv) > 1 else "probe.zarr"
ds = xr.open_zarr(path, consolidated=True)

a = ds["probe"].isel(time=0).values            # must be (lat, lon) = (180, 360)
assert a.shape == (180, 360), f"shape {a.shape} != (180, 360)"
assert ds["probe"].dims == ("time", "lat", "lon"), ds["probe"].dims

for iy, ix in [(0, 0), (44, 59), (89, 180), (179, 359)]:
    got = a[iy, ix]
    exp = iy * 1000 + ix
    assert got == exp, f"@(iy={iy}, ix={ix}): {got} != {exp}"   # north-up, lon-east, no transpose

assert np.isclose(ds.lat.values[0], 89.5) and np.isclose(ds.lat.values[-1], -89.5), ds.lat.values[[0, -1]]
assert np.isclose(ds.lon.values[0], -179.5) and np.isclose(ds.lon.values[-1], 179.5), ds.lon.values[[0, -1]]

print("dims:", ds["probe"].dims, "shape:", ds["probe"].shape, "encoding:", ds["probe"].encoding.get("chunks"))
print("PYTHON ORDERING OK")

File: test/zarr_probe/check.mjs
// Ordering check via zarrita (spec §4.3) — authoritative for the browser dashboard.
// Reads the store over HTTP through the consolidated metadata (.zmetadata), exactly
// as the front end does. zarrita 0.7 ships a default Blosc codec, so no manual
// codec registration is needed (numcodecs is pulled in transitively).
import * as zarr from "zarrita";

const url = process.argv[2] || "http://localhost:8123/probe.zarr";
const store = await zarr.withConsolidatedMetadata(new zarr.FetchStore(url));
const arr = await zarr.open(zarr.root(store).resolve("probe"), { kind: "array" });

const { data, shape } = await zarr.get(arr, [0, null, null]);   // Float32Array, shape [180, 360]
if (shape[0] !== 180 || shape[1] !== 360) {
  throw new Error(`JS shape FAIL: [${shape}] != [180,360]`);
}
const NX = 360;
for (const [iy, ix] of [[0, 0], [44, 59], [89, 180], [179, 359]]) {
  const got = data[iy * NX + ix];
  const exp = iy * 1000 + ix;
  if (got !== exp) throw new Error(`JS order FAIL @(${iy},${ix}): ${got} != ${exp}`);
}
console.log("JS ORDERING OK");
